packages/graphql/src/schema/generation/augment-where-input.ts

```diff
@@ -250,9 +269,15 @@ function fieldConfigsToFieldConfigMap({
     const fieldsConfigMap: InputTypeComposerFieldConfigMapDefinition = {};
 
     for (const field of fields) {
+        let directives;
+        if (deprecatedDirectives.length) {
+            directives = deprecatedDirectives;
+        } else if (field.deprecationReason) {
+            directives = [{ name: DEPRECATED, args: { reason: field.deprecationReason } }];
+        }
         fieldsConfigMap[field.name] = {
             type: field.typeName,
-            directives: deprecatedDirectives,
+            directives,
```

Comment:
Probably fine, but what happens if directives is undefined?

Reply:
From the typings it should not be possible!